TicketsDemo/App_Start/ReservationLoggingDecorator.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.App_Start
+{
+    public class ReservationLoggingDecorator : IReservationService
+    {
+        public IReservationService _decoratedObject;
+        private ILogger _logger;
+
+        public ReservationLoggingDecorator(IReservationService DecoratedObject, ILogger logger)
+        {
+            _decoratedObject = DecoratedObject;
+            _logger = logger;
+        }
+
+        public Reservation Reserve(PlaceInRun place)
+        {
+            try
+            {
+                var reserve = _decoratedObject.Reserve(place);
+                string message = string.Format("Successfully make a reservation for a ticket with Id {0}, number {2} for run {1} with run Id {3}.",
```

Comment:
in case of looooooong parameters string, it is more readable to put every parameter on it's own line